### DISCUSSION TITLE
Umi Max 我如何仅使用.env.local / .env.test / .env.prod 文件，就可以做到根据指令执行不同。在浏览器中可以访问不同的环境变量

### DISCUSSION THREAD

**Discussion by bnlwl** [umijs/umi#12327] [Q&A]
例如
```
# file .env.local

UMI_APP_URL=http://local:8000

```
```
# file .env.test

UMI_APP_URL=http://test:8000

```
```
# file .env.prod

UMI_APP_URL=http://prod:8000

```

指令是：
```
"scripts": {
    "build:prod": "max build",
    "build:test": "max build",
    "dev": "max dev",
  },
```

我期望：
执行 dev时，在浏览器中可以使用.env.local 中的变量
执行 build:test时，在浏览器中可以使用.env.test 中的变量
执行 build:prod时，在浏览器中可以使用.env.prod 中的变量

我现在的问题是。无论执行那个指令。浏览器中都.env.local中的变量

**Comment by bnlwl** [comment]
@fz6m 大佬~求助

**Reply by fz6m** [reply]
目前不支持这种 `.env.development` 的配置文件，在不同的环境对应的启动命令上用 `cross-env` 注入吧，或者定义在不同 `UMI_ENV` 的配置文件里。

参考 [`UMI_ENV`](https://umijs.org/docs/guides/env-variables#umi_env) ，不同的 `UMI_ENV` 分别会自动使用不同的配置文件，里面做值的区分即可（注意优先级顺序）。

**Reply by bnlwl** [reply]
好的。感谢大佬~

**Comment by rennzhang** [comment]
config.local.ts 的优先级并没有 config.prod.ts 高，要不是文档错误，要不是 bug，真坑

**Comment by xbsheng** [comment]
可以试下这个插件，能实现你需要的功能~  ： https://www.npmjs.com/package/@xbsheng/umi-plugin-env
